scipy/fftpack/logarithmic.py

```diff
@@ -0,0 +1,256 @@
+        (necessarily) the same quantity as kr.  rk is used only to multiply the
+        output array by sqrt(rk)^dir, so if you want to do the normalization
+        later, or you don't care about the normalization, you can set rk = 1.
+        Default is 1.
+
+    n : int, optional
+        Defines the length of the Fourier transform.  If `n` is not specified
+        (the default) then ``n = x.shape[axis]``.  If ``n < x.shape[axis]``,
+        `x` is truncated, if ``n > x.shape[axis]``, `x` is zero-padded.
+
+    axis : int, optional
+        The axis along which the transform is applied.  The default is the
+        last axis.
+
+    overwrite_x : bool, optional
+        If set to true, the contents of `x` can be overwritten. Default is
+        False.
+
+    Returns
+    -------
+    y : real ndarray
+        Transformed array Ã(k): a(j) is Ã(k_j) at k_j = k_c exp[(j-jc) dlnr].
+
+    .. versionadded:: 1.0.0
+
+    References
+    ----------
+    .. [1] 'Uncorrelated modes of the non-linear power spectrum', by A. J. S.
+           Hamilton, `Monthly Notices of the Royal Astronomical Society` vol.
+           312, pp. 257-284, http://dx.doy.org/10.1046/j.1365-8711.2000.03071.x
+           (2000). Website of FFTLog: http://casa.colorado.edu/~ajsh/FFTLog.
+
+    Examples
+    --------
+    >>> from scipy.fftpack import fftlog, fftlogargs
+    >>> # Get fftlog-arguments
+    >>> n, dlogr, logrc = 4, .1, 0
+    >>> q = 0
+    >>> mu = 'sine'
+    >>> w, t, kr, rk = fftlogargs(n, dlogr, logrc, mu, q, 1, 1)
+    >>> rk /= 2/np.pi    # Scale
+    >>> # Analytical solution
+    >>> fw = np.sqrt(np.pi/2/w)  # Frequency domain
+    >>> ft = 1/np.sqrt(t)        # Time domain
+    >>> # FFTLog
+    >>> fftl = fftlog(fw, dlogr=dlogr, mu=mu, q=q, kr=kr, rk=rk)
+    >>> fftl *= 2/np.pi  # Scale back
+    >>> # Print result
+    >>> print('Input      :', fw)
+    >>> print('Analytical :', ft)
+    >>> print('fftlog     :', fftl)
+    Input      : [ 1.48956664  1.32757767  1.18320484  1.05453243]
+    Analytical : [ 1.15380264  1.02832769  0.91649802  0.81682972]
+    fftlog     : [ 1.15380264  1.02832769  0.91649802  0.81682972]
+
+    """
+
+    # Check that mu is {'sine', or 'cosine'}
+    if mu not in ['sine', 'cosine']:
+        raise ValueError("mu must be either 'sine' or 'cosine'.")
+    if mu == 'sine':
+        nmu = 0.5
+    else:
+        nmu = -0.5
+
+    tmp = _asfarray(x)
+
+    if len(tmp) < 1:
+        raise ValueError("Invalid number of FFT data points "
+                         "(%d) specified." % len(tmp))
+
+    dlnr = dlogr*np.log(10.0)
+    n = len(tmp)
+    if np.iscomplexobj(tmp):  # Returns complex128
+        y = (_fftl.drfftl(tmp.real, n, nmu, q, dlnr, kr, rk, 1) +
+             1j*_fftl.drfftl(tmp.imag, n, nmu, q, dlnr, kr, rk, 1))
+    else:  # Returns float64
+        y = _fftl.drfftl(tmp, n, nmu, q, dlnr, kr, rk, 1)
+
+    return y
+
+
+def fftlogargs(n, dlogr=0.01, logrc=0.0, mu='sine', q=0, kr=1, kropt=0):
```

Comment:
Related to my API question for ``fftlog``: why is this function needed at all? Can you instead use this internally in ``fftlog``?